packages/react-native-web/src/modules/ImageLoader/index.js

```diff
@@ -111,37 +120,78 @@ const ImageLoader = {
       if (typeof failure === 'function') {
         failure();
       }
-      ImageLoader.abort(requestId);
+      ImageLoader.release(requestId);
       clearInterval(interval);
     }
   },
-  has(uri: string): boolean {
+  has(uri: ?string): boolean {
     return ImageUriCache.has(uri);
   },
-  load(uri: string, onLoad: Function, onError: Function): number {
+  load(
+    source: ImageSource,
+    onLoad: (ImageResult) => void,
+    onError: Function
+  ): number {
     id += 1;
     const image = new window.Image();
-    image.onerror = onError;
-    image.onload = (e) => {
+
+    const handleLoad = () => {
       // avoid blocking the main thread
-      const onDecode = () => onLoad({ nativeEvent: e });
-      if (typeof image.decode === 'function') {
-        // Safari currently throws exceptions when decoding svgs.
-        // We want to catch that error and allow the load handler
-        // to be forwarded to the onLoad handler in this case
-        image.decode().then(onDecode, onDecode);
-      } else {
-        setTimeout(onDecode, 0);
-      }
+      const onDecode = () => {
+        ImageUriCache.add(source.uri);
+        onLoad({
+          source: {
+            uri: image.src,
+            width: image.naturalWidth,
+            height: image.naturalHeight
+          }
+        });
```

Comment:
Made this match the RN `onLoad` interface 

https://github.com/Expensify/react-native-web/blob/7c6f58eaae57187b094364f9a30a1b40dd9b4855/packages/react-native-web/src/exports/Image/index.js#L374

This change is particularly useful in App here:
https://github.com/Expensify/App/pull/13036/files#diff-a501ab8bd2cd3a482e8ed66d2fe4f8736dc310280c2f639f6b9b1d4f529b96eaR16-R19

Otherwise to get `width` and `height` of an Image we need to call `getSize` and if you examine `ImageLoader.getSize` it has to fetch the image again and it doesn't work with headers (doesn't work with headers for RN as well for some reason) 

https://github.com/Expensify/react-native-web/blob/7c6f58eaae57187b094364f9a30a1b40dd9b4855/packages/react-native-web/src/modules/ImageLoader/index.js#L95-L109

---

This is a breaking change, previously `onLoad` was called with the Event object like this

```js
onLoad({ nativeEvent: e })
```

But this even object has nothing useful and it's very likely that noone uses it - it has no width/height props, it has no (direct) reference to the image node

Reply:
Here's what `onLoad` was previously called with 
<img width="224" alt="Screenshot 2023-01-05 at 14 36 47" src="https://user-images.githubusercontent.com/12156624/210781760-bc3bcd58-23b7-4897-ba1d-cf86df32e384.png">

Since `target`, `currentTarget` and `srcElement` are all `null` I see nothing useful for anybody to be using the data coming from the current `onLoad` implementation

_There's a way to get dimensions by traversing `e.path[0].naturalWidth`_

Reply:
The `nativeEvent` is still there

Previously `onLoad` was called with `nativeEvent` with no useful information (as the screenshot)
Now it is called with `nativeEvent` that contains a `source` object that contains width/height of the loaded image

---

Initially I just wanted to append `source` to the existing `nativeEvent` object, but since it's an HTML Event object it's nether possible to add fields to it or create a new object by spreading it

I guess if we want to go down this road we can return an object with all the Event fields and copy those values over and add `source` field with `width/height` as well, this way the change is non-breaking 
(We can always do that later if the maintainer would like that)

---

Unfortunately if we don't add this change App would not be able to make requests with headers

When we display images in App, right now we use `getSize` and do something with `width` and `height` then render the image
`getSize` does not work with headers, and though it's relatively easy to make it work with headers 
1. It won't match RN's interface
2. It's another request to make vs making width/height available from the `onLoad` event